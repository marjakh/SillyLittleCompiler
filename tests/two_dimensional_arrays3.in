let x = 2;
let y = 3;

let a = new Array(x);
let i = 0;
let j = 0;
while (i < x) {
  a[i] = new Array(y);
  i = i + 1;
}
i = 0;
while (i < x) {
  j = 0;
  while (j < y) {
    a[i][j] = i * 10 + j;
    write(i);
    write(j);
    write(a[i][j]);
   j = j + 1;
  }
  i = i + 1;
}

i = 0;
while (i < x) {
  j = 0;
  while (j < y) {
    write(i);
    write(j);
    write(a[i][j]);
    j = j + 1;
  }
  i = i + 1;
}
