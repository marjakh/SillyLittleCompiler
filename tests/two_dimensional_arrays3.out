0
0
0
0
1
1
0
2
2
1
0
10
1
1
11
1
2
12
0
0
0
0
1
1
0
2
2
1
0
10
1
1
11
1
2
12
